src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h

```diff
@@ -1708,7 +1788,176 @@ namespace UiaOperationAbstraction
+        {
+        }
+
+        UiaTuple(ItemWrapperType... args) : UiaTuple()
+        {
+            SetTupleWrapperItems<0 /* I */>(args...);
+        }
+
+        UiaTuple(winrt::Microsoft::UI::UIAutomation::AutomationRemoteArray array) :
+            UiaTypeBase(array)
+        {
+        }
+
+        explicit UiaTuple(winrt::Microsoft::UI::UIAutomation::AutomationRemoteAnyObject remoteValue) :
+            UiaTypeBase(remoteValue.AsArray())
+        {
+        }
+
+        UiaTuple(const UiaTuple&) = default;
+
+        UiaBool IsNull() const
+        {
+            if (ShouldUseRemoteApi())
+            {
+                auto remoteValue = std::get_if<RemoteType>(&m_member);
+                if (remoteValue)
+                {
+                    return remoteValue->IsNull();
+                }
+            }
+
+            // The local version of a UiaTuple can never be null.
+            return false;
+        }
+
+        UiaBool operator!() const { return IsNull(); }
+        operator UiaBool() const { return !IsNull(); }
+
+        operator std::shared_ptr<TupleLocalType>() const
+        {
+            return std::get<std::shared_ptr<TupleLocalType>>(m_member);
+        }
+
+        TupleLocalType& operator*()
+        {
+            return *std::get<LocalType>(m_member);
+        }
+
+        const TupleLocalType& operator*() const
+        {
+            return *std::get<LocalType>(m_member);
+        }
+
+        UiaBool operator==(const UiaTuple& rhs) const
+        {
+            if (ShouldUseRemoteApi())
+            {
+                auto mutableThis = *this;
+                mutableThis.ToRemote();
+
+                auto mutableRhs = rhs;
+                mutableRhs.ToRemote();
+                return std::get<RemoteType>(mutableThis.m_member).IsEqual(std::get<RemoteType>(mutableRhs.m_member));
+            }
+
+            auto localTuple = std::get<LocalType>(m_member);
+            auto localRhsTuple = std::get<LocalType>(rhs.m_member);
+            return *localTuple == *localRhsTuple;
+        }
+
+        UiaBool operator!=(const UiaTuple& rhs) const
+        {
+            return !(*this == rhs);
+        }
+
+        void ToRemote()
+        {
+            auto delegator = UiaOperationScope::GetCurrentDelegator();
+            if (delegator)
+            {
+                delegator->ConvertVariantDataToRemote<ItemWrapperType...>(m_member);
+            }
+        }
+
+        template <unsigned int I>
+        void SetAt(TupleItemWrapperType<I> item)
+        {
+            if (ShouldUseRemoteApi())
+            {
+                UiaUint index{ I };
+                ToRemote();
+                index.ToRemote();
+                item.ToRemote();
+                std::get<RemoteType>(m_member).SetAt(index, item);
+                return;
+            }
+
+            std::get<I>(*std::get<LocalType>(m_member)) = item;
+        }
+

```

Comment:
Did we remove the `Size()` function?

Reply:
Yes. `Size` makes sense in case of dynamic collections. The type of tuple we have here is not dynamic in the sense of contained values (and index-based access through abstraction types is -- to an extent -- protected against accessing outside the tuple size definition through the tools that `std::tuple` provides).